### ISSUE TITLE
Bidseditor crashes

### ISSUE THREAD

**Issue by pimpul** [Donders-Institute/bidscoin#59]
Hello,

When trying to edit the bidsmap in the editor, it crashes. command used:  

```console
~$ bidsmapper . -n pt organized/ -i 2

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/bidscoin/bidseditor.py", line 772, in handle_edit_button_clicked
    modality   = self.samples_table.item(rowindex, 2).text()
AttributeError: 'NoneType' object has no attribute 'text'
Abort trap: 6
```
Mac OSX 10.14.6
Python 3.9.0

**Comment by marcelzwiers** [comment]
There must be something in your bidsmap.yaml file that I haven't encountered before. Can you send me your `organized/code/bidscoin/bidsmap.yaml` file so I can have a look?

**Comment by pimpul** [comment]
Hi Marcel,

some background: I want to organize SWI data as anat/T2*. 

thanks,
 Pim

# --------------------------------------------------------------------------------
# This is a bidsmap YAML file with the key-value mappings for the different BIDS
# modalities (anat, func, dwi, etc). The modality attributes are the keys that map
# onto the BIDS labels. The bidsmap data-structure should be 5 levels deep:
# dict > dict > list > dict > dict
#
# NB:
# 1) Edit the bidsmap file to your needs before feeding it to bidscoiner.py
# 2) (Institute) users may create their own bidsmap_[template].yaml or
#    bidsmap_[sample].yaml file
#
# For more information, see:
# https://github.com/Donders-Institute/bidscoin
# https://docs.ansible.com/ansible/latest/reference_appendices/YAMLSyntax.html
# --------------------------------------------------------------------------------


Options:
  bidscoin:
    version: 3.0.8                # BIDScoin version (should correspond with the version in ../bidscoin/version.txt)
    bidsignore: extra_data/       # Semicolon-separated list of entries that are added to the .bidsignore file (for more info, see BIDS specifications), e.g. extra_data/;pet/;myfile.txt;yourfile.csv
  dcm2niix:                       # See dcm2niix -h and https://www.nitrc.org/plugins/mwiki/index.php/dcm2nii:MainPage#General_Usage for more info
    path: module add dcm2niix;    # Command to set the path to dcm2niix (note the semi-colon), e.g. module add dcm2niix/1.0.20180622; or PATH=/opt/dcm2niix/bin:$PATH; or /opt/dcm2niix/bin/ or '"C:\Program Files\dcm2niix\"' (note the quotes to deal with the whitespace)
    args: -b y -z y -i n          # Argument string that is passed to dcm2niix. Tip: SPM users may want to use '-z n' (which produces unzipped nifti's, see dcm2niix -h for more information)


PlugIns: []
DICOM:
  subject: <<SourceFilePath>>     # <<SourceFilePath>> extracts the subject label from the source directory during bidscoiner runtime. A DICOM attribute can also be used as subject-label instead, e.g. <PatientID>
  session: <<SourceFilePath>>     # <<SourceFilePath>> extracts the session label from the source directory during bidscoiner runtime. A DICOM attribute can also be used as session-label instead, e.g. <StudyID>
  anat:       # ----------------------- All anatomical runs --------------------
  func:       # ----------------------- All functional runs --------------------
  dwi:        # ----------------------- All diffusion runs ---------------------
  fmap:       # ----------------------- All fieldmap runs ----------------------
  beh:        # ----------------------- All behavioural data -------------------
  pet:        # ----------------------- All PET runs ---------------------------
  extra_data: # ----------------------- All extra data -------------------------
  - provenance: /Users/pimpullens/DATA/currentProjects/HAPCI/swi/DATA/microbleeds/AMC_Anonymous/pt100_mm0/00701_SWIp_fast/00001.dcm
    attributes:
      Modality: MR
      ProtocolName: SWIp_fast SENSE
      SeriesDescription: SWIp_fast
      ImageType: "['ORIGINAL', 'PRIMARY', 'SW_M_FFE', 'SW_M', 'FFE']"
      SequenceName: ''
      SequenceVariant: SP
      ScanningSequence: GR
      MRAcquisitionType: 3D
      SliceThickness: '2'
      FlipAngle: '17'
      EchoNumbers: 1
      EchoTime: '0'
      RepetitionTime: '31'
      PhaseEncodingDirection: ''
    bids:
      acq: SWIpfast
      rec:
      ce:
      task:
      echo: '1'
      dir: ROW
      run: <<1>>
      mod:
      suffix:
  - provenance: /Users/pimpullens/DATA/currentProjects/HAPCI/swi/DATA/microbleeds/AMC_Anonymous/pt100_mm1/00701_SWIp_fast/00001.dcm
    attributes:
      Modality: MR
      ProtocolName: SWIp_fast
      SeriesDescription: SWIp_fast
      ImageType: "['ORIGINAL', 'PRIMARY', 'SW_M_FFE', 'SW_M', 'FFE']"
      SequenceName: ''
      SequenceVariant: SP
      ScanningSequence: GR
      MRAcquisitionType: 3D
      SliceThickness: '2'
      FlipAngle: '17'
      EchoNumbers: 1
      EchoTime: '0'
      RepetitionTime: '31'
      PhaseEncodingDirection: ''
    bids:
      acq: SWIpfast
      rec:
      ce:
      task:
      echo: '1'
      dir: ROW
      run: <<1>>
      mod:
      suffix:
  - provenance: /Users/pimpullens/DATA/currentProjects/HAPCI/swi/DATA/microbleeds/AMC_Anonymous/pt108_mm0/00701_SWIp_fast/00001.dcm
    attributes:
      Modality: MR
      ProtocolName: WIP SWIp_fast
      SeriesDescription: SWIp_fast
      ImageType: "['ORIGINAL', 'PRIMARY', 'SW_M_FFE', 'SW_M', 'FFE']"
      SequenceName: ''
      SequenceVariant: SP
      ScanningSequence: GR
      MRAcquisitionType: 3D
      SliceThickness: '2'
      FlipAngle: '17'
      EchoNumbers: 1
      EchoTime: '0'
      RepetitionTime: '31'
      PhaseEncodingDirection: ''
    bids:
      acq: SWIpfast
      rec:
      ce:
      task:
      echo: '1'
      dir: ROW
      run: <<1>>
      mod:
      suffix:
  - provenance: /Users/pimpullens/DATA/currentProjects/HAPCI/swi/DATA/microbleeds/AMC_Anonymous/pt108_mm0/00702_SWI/00001.dcm
    attributes:
      Modality: MR
      ProtocolName: SWI
      SeriesDescription: SWI
      ImageType: "['ORIGINAL', 'PRIMARY', 'PROJECTION IMAGE', 'SW_M', 'FFE']"
      SequenceName: ''
      SequenceVariant: SP
      ScanningSequence: GR
      MRAcquisitionType: 3D
      SliceThickness: '3'
      FlipAngle: '17'
      EchoNumbers: 1
      EchoTime: '0'
      RepetitionTime: '31'
      PhaseEncodingDirection: ''
    bids:
      acq: SWI
      rec:
      ce:
      task:
      echo: '1'
      dir: <InPlanePhaseEncodingDirection>
      run: <<1>>
      mod:
      suffix:
  - provenance: /Users/pimpullens/DATA/currentProjects/HAPCI/swi/DATA/microbleeds/AMC_Anonymous/pt44_mm1/00701_SWIp_fast/00001.dcm
    attributes:
      Modality: MR
      ProtocolName: WIP SWIp_fast SENSE
      SeriesDescription: SWIp_fast
      ImageType: "['ORIGINAL', 'PRIMARY', 'SW_M_FFE', 'SW_M', 'FFE']"
      SequenceName: ''
      SequenceVariant: SP
      ScanningSequence: GR
      MRAcquisitionType: 3D
      SliceThickness: '2'
      FlipAngle: '17'
      EchoNumbers: 1
      EchoTime: '0'
      RepetitionTime: '31'
      PhaseEncodingDirection: ''
    bids:
      acq: SWIpfast
      rec:
      ce:
      task:
      echo: '1'
      dir: ROW
      run: <<1>>
      mod:
      suffix:
  leave_out:  # ----------------------- Data that will be left out -------------
PAR:
  subject: <<SourceFilePath>>     # <<SourceFilePath>> extracts the subject label from the source directory during bidscoiner runtime. A DICOM attribute can also be used as subject-label instead, e.g. <PatientID>
  session: <<SourceFilePath>>     # <<SourceFilePath>> extracts the session label from the source directory during bidscoiner runtime. A DICOM attribute can also be used as session-label instead, e.g. <StudyID>
  anat:       # ----------------------- All anatomical runs --------------------
  func:       # ----------------------- All functional runs --------------------
  dwi:        # ----------------------- All diffusion runs ---------------------
  fmap:       # ----------------------- All fieldmap runs ----------------------
  beh:        # ----------------------- All behavioural data -------------------
  pet:        # ----------------------- All PET runs ---------------------------
  extra_data: # ----------------------- All extra data -------------------------
  leave_out:  # ----------------------- Data that will be left out -------------
P7:
# --------------------------------------------------------------------------------
# P*.7 key-value heuristics (GE fields that are mapped to the BIDS labels)
# --------------------------------------------------------------------------------


Nifti:
# --------------------------------------------------------------------------------
# Nifti key-value heuristics (Nifti fields that are mapped to the BIDS labels)
# --------------------------------------------------------------------------------


FileSystem:
  subject: <<SourceFilePath>>       # <<SourceFilePath>> extracts the subject label from the source directory during bidscoiner runtime
  session: <<SourceFilePath>>       # <<SourceFilePath>> extracts the session label from the source directory during bidscoiner runtime
  anat:       # ----------------------- All anatomical runs --------------------
  func:       # ----------------------- All functional runs --------------------
  dwi:        # ----------------------- All diffusion runs ---------------------
  fmap:       # ----------------------- All fieldmap runs ----------------------
  beh:        # ----------------------- All behavioural data -------------------
  extra_data: # ----------------------- All extra data -------------------------
  leave_out:  # ----------------------- Data that will be left out -------------

**Comment by marcelzwiers** [comment]
Can you either attach the bidsmap.yaml file or edit your message and put these two code-block lines?
\`\`\`yaml
[your bidsmap content]
\`\`\`

**Comment by marcelzwiers** [comment]
Mhhh, I can edit your bidsmap just fine here on my Linux machine. I don't have a Mac to test the GUI, but it could perhaps be that the QTableWidget of the PyQT library doesn't properly read out the mouse position OSX? Do you have the possibility to test this under e.g. Linux?

**Comment by marcelzwiers** [comment]
Mhhh, when I upgrade to the latest PyQT5 library I also get some errors. Could you try to run this in your bidscoin python environment and then try again?
```console
pip uninstall pyqt5
pip install pyqt5==5.12.2
```

**Comment by marcelzwiers** [comment]
Fixed with commit a49beb1c04cd8ab963782a40ad1ed5081a9fce8a
